solr/core/src/test/org/apache/solr/cloud/overseer/ZkStateReaderTest.java

```diff
@@ -196,4 +209,251 @@ public void testWatchedCollectionCreation() throws Exception {
       server.shutdown();
     }
   }
+
+  public void testForciblyRefreshAllClusterState() throws Exception {
+    Path zkDir = createTempDir("testForciblyRefreshAllClusterState");
+
+    ZkTestServer server = new ZkTestServer(zkDir);
+
+    SolrZkClient zkClient = null;
+    ZkStateReader reader = null;
+
+    try {
+      server.run();
+
+      zkClient = new SolrZkClient(server.getZkAddress(), OverseerTest.DEFAULT_CONNECTION_TIMEOUT);
+      ZkController.createClusterZkNodes(zkClient);
+
+      reader = new ZkStateReader(zkClient);
+      reader.createClusterStateWatchersAndUpdate();
+      reader.registerCore("c1"); // watching c1, so it should get non lazy reference
+      zkClient.makePath(ZkStateReader.COLLECTIONS_ZKNODE + "/c1", true);
+
+      reader.forciblyRefreshAllClusterStateSlow();
+      // Initially there should be no c1 collection.
+      assertNull(reader.getClusterState().getCollectionRef("c1"));
+
+      ZkStateWriter writer = new ZkStateWriter(reader, new Stats());
```

Comment:
Can we pull this into a common setup method so the logic isn't repeated in every single test method?

Reply:
Nice suggestion! I'm on it now!